File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i = 0, f = 0;
        int l = s.size();
        string t;
        for (i = 0; i < l; i++) {
            if (isalnum(s[i])) {
                t += s[i];
            }
        }
        int l2 = t.size();
        for (i = 0; i < l2 / 2; i++) {
            if (tolower(t[i]) == tolower(t[l2 - i - 1]))
                continue;
            else {
                f = 1;
                break;
            }
        }
        if (f == 1)
            return false;
        else
            return true;
            
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean checkPalin(String cs,int i){
        if(i>=cs.length()/2){
            return true;
        }
        if(cs.charAt(i)!=cs.charAt(cs.length()-i-1)){
            return false;
        }
        return checkPalin(cs,i+1);
    }
    public boolean isPalindrome(String s) {
        String cs="";
        for(int i=0;i<s.length();i++){
            char ch=s.charAt(i);
            if(Character.isLetterOrDigit(ch)){
                cs=cs+Character.toLowerCase(ch);
            }
        }
        return checkPalin(cs,0);
    }
}